src/App.tsx

```diff
@@ -1,24 +1,203 @@
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deleteTodoId, setDeleteTodoId] = useState(0);
+
+  const activeTodo = todos.filter(todo => !todo.completed).length;
+  const completedTodo = todos.filter(todo => todo.completed).length;
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((res) => {
+        setTodos(res);
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+
+        const timer = setTimeout(() => {
+          setError('');
+        }, 3000);
+
+        return () => {
+          clearTimeout(timer);
+        };
+      });
+  }, []);
+
+  const addTodo = (title: string) => {
+    if (!title.trim()) {
+      setError('Title can`t be empty');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } else {
+      const newTodo = {
+        id: 0,
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+      setIsDisabled(true);
+      setError('');
+
+      createTodos(USER_ID, newTodo)
+        .then((res) => {
+          setTodos((prevTodo) => {
+            return [...prevTodo, res];
+          });
+        })
+        .catch(() => {
+          setError('Unable to add a todo');
+
+          const timer = setTimeout(() => {
+            setError('');
+          }, 3000);
+
+          return () => {
+            clearTimeout(timer);
+          };
+        })
+        .finally(() => {
+          setIsDisabled(false);
+          setTempTodo(null);
+        });
+    }
+  };
+
+  const deleteTodo = (id: number) => {
+    setDeleteTodoId(id);
+
+    removeTodo(id)
+      .then(() => {
+        const result = todos.filter(todo => todo.id !== id);
+
+        setTodos(result);
+        setError('');
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+
+        const timer = setTimeout(() => {
+          setError('');
+        }, 3000);
+
+        return () => {
+          clearTimeout(timer);
+        };
+      })
+      .finally(() => {
+        setDeleteTodoId(null || 0);
+      });
+  };
+
+  const handleChangeCompleted = (id: number) => {
+    const completed = todos.map(item => (
+      item.id === id
+        ? {
+          ...item,
+          completed: !item.completed,
+        }
+        : item
+    ));
+
+    setTodos(completed);
+  };
+
+  const handleClearCompleted = () => {
+    const filteredTodos = todos.filter(item => !item.completed);
+
+    todos.forEach(item => {
+      if (item.completed) {
+        removeTodo(item.id);
+      }
+    });
+
+    setTodos(filteredTodos);
+  };
+
+  const handleToggleAll = () => {
+    setIsActive(!isActive);
+  };
+
+  useEffect(() => {
+    const result = todos.map(item => {
+      return { ...item, completed: isActive };
+    });
+
+    setTodos(result);
+  }, [isActive]);
+
+  const filteredTodos = useMemo(() => {
+    return todos?.filter(todo => {
```

Comment:
Todos can't be null or undefined
```suggestion
    return todos.filter(todo => {
```